src/replica/HttpIngestTransModule.cc

```diff
@@ -257,124 +270,179 @@ json HttpIngestTransModule::_beginTransaction() {
 json HttpIngestTransModule::_endTransaction() {
     debug(__func__);
 
-    TransactionId id = 0;
-    string database;
-    bool abort = false;
-
-    auto const logEndTransaction = [&](string const& status, string const& msg=string()) {
-        ControllerEvent event;
-        event.operation = "END TRANSACTION";
-        event.status = status;
-        event.kvInfo.emplace_back("id", to_string(id));
-        event.kvInfo.emplace_back("database", database);
-        event.kvInfo.emplace_back("abort", abort ? "true" : "false");
-        if (not msg.empty()) event.kvInfo.emplace_back("error", msg);
-        logEvent(event);
-    };
-
-    // Intercept standard exceptions just to report the failure, then
-    // let a caller to do the rest (post error messages into the Logger,
-    // return errors to clients).
-    try {
-        auto const config = controller()->serviceProvider()->config();
-        auto const databaseServices = controller()->serviceProvider()->databaseServices();
+    auto const config = controller()->serviceProvider()->config();
+    auto const databaseServices = controller()->serviceProvider()->databaseServices();
 
-        id = stoul(params().at("id"));
-        abort = query().requiredBool("abort");
+    TransactionId const id = stoul(params().at("id"));
```

Comment:
any possible conversion error is handled above, or known to not happen at this point?